src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.WorkspaceHost.cs

```diff
@@ -16,56 +17,49 @@ internal partial class ServiceHubRemoteHostClient
         private class WorkspaceHost : ForegroundThreadAffinitizedObject, IVisualStudioWorkspaceHost, IVisualStudioWorkingFolder
         {
             private readonly VisualStudioWorkspaceImpl _workspace;
-            private readonly RemoteHostClient _client;
+
+            /// <summary>
+            /// The current session we have open to the remote host.  Only accessible from the
+            /// UI thread.
+            /// </summary>
+            private readonly KeepAliveSession _session;
```

Comment:
I dont think WorkspaceHost will go away until VS close. so I dont think there is lifetime issue. but if you want to explicitly control it. you can call Shutdown on KeepAliveSession when VS goes away.